src/components/Search/index.tsx

```diff
@@ -0,0 +1,214 @@
+import { useEffect, useRef, useState } from 'react'
+import Image from 'next/image'
+import Link from 'next/link'
+
+import { searchClient } from 'utils/meilisearchClient'
+import {
+  InstantSearch,
+  Highlight,
+  Configure,
+  connectStateResults,
+  connectSearchBox,
+  connectHits
+} from 'react-instantsearch-dom'
+import { getImageUrl } from 'utils/getImageUrl'
+import formatPrice from 'utils/format-price'
+
+import { GameFragment_cover } from 'graphql/generated/GameFragment'
+
+import {
+  Search as SearchIcon,
+  Close as CloseIcon
+} from '@styled-icons/material-outlined'
+import { Apple, Windows, Linux } from '@styled-icons/fa-brands'
+import * as S from './styles'
+import Button from 'components/Button'
+
+const Search = () => {
```

Comment:
I'd separate those parts in different components (maybe even internal ones) like:

```
- Search
  - SearchBox
  - Hits
  - NoResults
```

Reply:
Agree. I'll break it up into smaller components as you suggest

Reply:
hey @willianjusten just to check before pushing changes, do you think this structure is valid/makes sense?

![shot_210630_124245](https://user-images.githubusercontent.com/50624358/123991451-1af29680-d9a1-11eb-9067-063d8c27f57a.png)

Reply:
Yeah, that was exactly what I thought. Since they're dependant, doesn't make sense to have it separated but together like this.